src/sentry/integrations/msteams/webhook.py

```diff
@@ -197,10 +225,125 @@ class MsTeamsWebhookEndpoint(Endpoint, MsTeamsWebhookMixin):
     permission_classes = ()
     provider = "msteams"
 
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self._event_handlers: dict[MsTeamsEvents, Callable[[HttpRequest], HttpResponse]] = {
+            MsTeamsEvents.MESSAGE: self.handle_message_event,
+            MsTeamsEvents.CONVERSATION_UPDATE: self.handle_conversation_update_event,
+            MsTeamsEvents.UNKNOWN: self.handle_unknown_event,
+        }
+
     @csrf_exempt
     def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         return super().dispatch(request, *args, **kwargs)
 
+    @classmethod
+    def _get_team_installation_request_data(cls, data: dict[str, Any]) -> dict:
+        """
+        Helper method that will construct the installation request for a MsTeams team channel.
+        We want the KeyError exception to be raised if the key does not exist.
+        """
+        channel_data = data["channelData"]
+        new_team_info = channel_data["team"]
+
+        team_id = new_team_info.get("aadGroupId", None)
+        if team_id is None:
+            logger.info(
+                "sentry.integrations.msteams.webhooks: New team info data does not have aadGroupId",
+                extra={"data": data},
+            )
+            fallback_id = new_team_info["id"]
+            team_id = fallback_id
+
+        team_name = new_team_info["name"]
+        service_url = data["serviceUrl"]
+        from_data = data["from"]
+        user_id = from_data["id"]
+
+        tenant_info = channel_data["tenant"]
+        tenant_id = tenant_info["id"]
+        params = {
+            "service_url": service_url,
+            "user_id": user_id,
+            "tenant_id": tenant_id,
+            "conversation_id": team_id,
+            "external_id": team_id,
+            "external_name": team_name,
+            "installation_type": "team",
+        }
+        return params
+
+    def handle_installation_update_event(self, request: HttpRequest) -> HttpResponse:
+        data = request.data
+        action = data.get("action", None)
+        if action is None or action != "add":
+            logger.info(
+                "sentry.integrations.msteams.webhooks: Action not supported",
+                extra={"request_data": data},
+            )
+            return self.respond(status=204)
+
+        try:
+            installation_params = self._get_team_installation_request_data(data=data)
+        except Exception as err:
+            logger.info(
+                "sentry.integrations.msteams.webhooks: Installation param error",
+                exc_info=err,
+                extra={"request_data": data},
+            )
+            return self.respond(status=400)
```

Comment:
should we respond with an error message here?

Reply:
I don't know what MSTeams will show or do with the error :(
But we can!

Reply:
done!